### ISSUE TITLE
`build/gulpfile.editor.js` is not searched in our repository

### ISSUE THREAD

**Issue by alexdima** [microsoft/vscode#107356]
* search for `editor-esm-bundle`
* observe the matches in `build/gulpfile.editor.js` are not found.
* I don't think this file should be excluded based on the settings:
```
	"search.exclude": {
		"**/node_modules": true,
		"**/bower_components": true,
		".build/**": true,
		"out/**": true,
		"out-build/**": true,
		"out-vscode/**": true,
		"i18n/**": true,
		"extensions/**/out/**": true,
		"test/smoke/out/**": true,
		"test/automation/out/**": true,
		"test/integration/browser/out/**": true,
		"src/vs/base/test/node/uri.test.data.txt": true,
		"src/vs/workbench/test/browser/api/extHostDocumentData.test.perf-data.ts": true,
		"src/vs/server": false
	},
```

**Comment by roblourens** [comment]
It's in `build/.gitignore` since 2 days ago @joaomoreno

**Comment by alexdima** [comment]
wtf

**Comment by alexdima** [comment]
Interestingly, when I touch the file it appears as modified, so I guess it is not `.gitignored` ?

![image](https://user-images.githubusercontent.com/5047891/94159063-69f42400-fe83-11ea-80c5-5ffd36398a85.png)

**Comment by roblourens** [comment]
It's a file that's checked in to git, but in gitignore, this is a valid thing to do for git. But ripgrep only knows about the `.gitignore` file. I would agree that this is a little unexpected for users but I don't think it's practical for rg to query git during a search.

**Comment by alexdima** [comment]
I've tried to tackle this by having as a goal to put all `.ts` files and only `.ts` files in `build/lib`:
* converted all `.js` files from `build/lib` to `.ts`.
* moved more of the `.ts` files to `build/lib`

There are still 2 folders left:
* `azure-pipelines`
* `darwin`

Since `darwin` contains only `.ts` files and `azure-pipelines` also almost exclusively `.ts` files, the final ToDo here is to convert `build/azure-pipelines/mixin.js` and `build/azure-pipelines/upload-sourcemaps.js` to TS. That would make the `.gitignore` align with the reality of the repository and would make search 100% reliant again.

I've converted a bunch of files, and will take a break now.
